src/metabase/db/custom_migrations.clj

```diff
@@ -1255,13 +1256,27 @@
     (qs/shutdown scheduler)))
 
 (define-migration BackfillQueryField
-  (let [update-query-fields! (requiring-resolve 'metabase.native-query-analyzer/update-query-fields-for-card!)
-        cards                (t2/select :model/Card :id [:in {:from      [[:report_card :c]]
-                                                              :left-join [[:query_field :f] [:= :f.card_id :c.id]]
-                                                              :select    [:c.id]
-                                                              :where     [:and
-                                                                          [:not :c.archived]
-                                                                          [:= :c.query_type "native"]
-                                                                          [:= :f.id nil]]}])]
-    (doseq [card cards]
-      (update-query-fields! card))))
+  (let [field-ids-for-sql  (requiring-resolve 'metabase.native-query-analyzer/field-ids-for-sql)
+        ;; practically #'metabase.models.query-field/field-ids-for-mbql
+        field-ids-for-mbql (fn [query]
+                             {:direct (some-> (lib.util.match/match query ["field" (id :guard integer?) _] id)
```

Comment:
I think we can use `clojure.core/match` to minimize dependencies here and avoid tying `lib.util.match` to a migration. 

Also this will miss the field IDs used in implicit joins, e.g. 
```
["field", 6, {base-type: "type/Text", source-field: 5}]
```